pkg/spi/resource/spi.go

```diff
@@ -0,0 +1,49 @@
+package resource
+
+import (
+	"encoding/json"
+
+	"github.com/docker/infrakit/pkg/spi"
+)
+
+// InterfaceSpec is the current name and version of the resource API.
+var InterfaceSpec = spi.InterfaceSpec{
+	Name:    "Resource",
+	Version: "0.1.0",
+}
+
+// Plugin is a vendor-agnostic API used to create and manage resources with an infrastructure provider.
+type Plugin interface {
```

Comment:
I'm just wandering in here but one idea I'd had about this kind of thing historically is to have a few different `interface`s which dictate the behavior of various resources.  For instance, the vast majority of instances can be `Create`-ed and `Destroy`-ed, but only some things (e.g., instances) can be `Stop`-ed and `Start`-ed...

Maybe your RPC server could accept attempted calls on a particular resource and throw issue if that particular plugin resource doesn't embed the given interface...

e.g.,

```go
// Create, Read, Delete
type CRDer interface {
    Create() error
    Read() (json.RawMessage, error)
    Delete() error
}

type StartStopper interface {
    Start() error
    Stop() error
}

type Instance interface {
    CRUDer
    StartStopper
}

type SecurityGroup interface {
    CRUDer
}
```

(It's worth considering more uncommon verbs like `Attach` as well... You can actually `Attach` and `Detach` security groups from an instance.  That's a totally different operation than just `Create`-ing some with given properties and could end up being pretty relevant for, e.g., VPC peering).